subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/builder/DependencyState.java

```diff
@@ -146,9 +145,6 @@ void addSelectionReasons(List<ComponentSelectionDescriptorInternal> reasons) {
         if (!ruleDescriptors.isEmpty()) {
             addRuleDescriptors(reasons);
         }
-        if (isDependencyForced()) {
```

Comment:
There are still ways to "force" a node, just not on the node itself. So this remains a possible state.